core/flyout_base.js

```diff
@@ -466,7 +482,22 @@ Blockly.Flyout.prototype.show = function(xmlList) {
       var tagName = xml.tagName.toUpperCase();
       var default_gap = this.horizontalLayout_ ? this.GAP_X : this.GAP_Y;
       if (tagName == 'BLOCK') {
-        var curBlock = Blockly.Xml.domToBlock(xml, this.workspace_);
+
+        // We assume that in a flyout, the same block id/type means
+        // the same output SVG nodes.  Check for a recycled element.
+        var id = xml.getAttribute('id') || xml.getAttribute('type');
+        var recycled = this.recycleBlocks_.findIndex(function(block) {
+          return block.id === id;
+        });
+        var curBlock;
+
+        if (recycled > -1) {
+          // Delete the recycled element out of the recycle bin and
+          // assign it to curBlock.
+          curBlock = this.recycleBlocks_.splice(recycled, 1)[0];
```

Comment:
@kchadha @paulkaplan Recycling a block this way means that there will be no delete or create events fired.  Will the VM handle that correctly?